### ISSUE TITLE
ImportError: /usr/local/lib/python3.10/dist-packages/torch/lib/../../nvidia/cusparse/lib/libcusparse.so.12: undefined symbol: __nvJitLinkAddData_12_1, version libnvJitLink.so.12

### ISSUE THREAD

**Issue by qianxifu** [pytorch/pytorch#111469]
### 🐛 Describe the bug

```
>>> import torch
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/usr/local/lib/python3.10/dist-packages/torch/__init__.py", line 235, in <module>
    from torch._C import *  # noqa: F403
ImportError: /usr/local/lib/python3.10/dist-packages/torch/lib/../../nvidia/cusparse/lib/libcusparse.so.12: undefined symbol: __nvJitLinkAddData_12_1, version libnvJitLink.so.12
>>> 
```

### Versions

Collecting environment information...
PyTorch version: N/A
Is debug build: N/A
CUDA used to build PyTorch: N/A
ROCM used to build PyTorch: N/A

OS: Ubuntu 20.04.6 LTS (x86_64)
GCC version: (Ubuntu 9.4.0-1ubuntu1~20.04.2) 9.4.0
Clang version: Could not collect
CMake version: Could not collect
Libc version: glibc-2.31

Python version: 3.10.13 (main, Aug 25 2023, 13:20:03) [GCC 9.4.0] (64-bit runtime)
Python platform: Linux-5.4.0-162-generic-x86_64-with-glibc2.31
Is CUDA available: N/A
CUDA runtime version: 12.0.140
CUDA_MODULE_LOADING set to: N/A
GPU models and configuration: GPU 0: NVIDIA A10
Nvidia driver version: 525.105.17
cuDNN version: Probably one of the following:
/usr/local/cuda-12.0/targets/x86_64-linux/lib/libcudnn.so.8.9.1
/usr/local/cuda-12.0/targets/x86_64-linux/lib/libcudnn_adv_infer.so.8.9.1
/usr/local/cuda-12.0/targets/x86_64-linux/lib/libcudnn_adv_train.so.8.9.1
/usr/local/cuda-12.0/targets/x86_64-linux/lib/libcudnn_cnn_infer.so.8.9.1
/usr/local/cuda-12.0/targets/x86_64-linux/lib/libcudnn_cnn_train.so.8.9.1
/usr/local/cuda-12.0/targets/x86_64-linux/lib/libcudnn_ops_infer.so.8.9.1
/usr/local/cuda-12.0/targets/x86_64-linux/lib/libcudnn_ops_train.so.8.9.1
HIP runtime version: N/A
MIOpen runtime version: N/A
Is XNNPACK available: N/A

CPU:
Architecture:                       x86_64
CPU op-mode(s):                     32-bit, 64-bit
Byte Order:                         Little Endian
Address sizes:                      46 bits physical, 48 bits virtual
CPU(s):                             16
On-line CPU(s) list:                0-15
Thread(s) per core:                 2
Core(s) per socket:                 8
Socket(s):                          1
NUMA node(s):                       1
Vendor ID:                          GenuineIntel
CPU family:                         6
Model:                              106
Model name:                         Intel(R) Xeon(R) Platinum 8369B CPU @ 2.90GHz
Stepping:                           6
CPU MHz:                            2900.000
BogoMIPS:                           5800.00
Hypervisor vendor:                  KVM
Virtualization type:                full
L1d cache:                          384 KiB
L1i cache:                          256 KiB
L2 cache:                           10 MiB
L3 cache:                           48 MiB
NUMA node0 CPU(s):                  0-15
Vulnerability Gather data sampling: Unknown: Dependent on hypervisor status
Vulnerability Itlb multihit:        Not affected
Vulnerability L1tf:                 Not affected
Vulnerability Mds:                  Not affected
Vulnerability Meltdown:             Not affected
Vulnerability Mmio stale data:      Vulnerable: Clear CPU buffers attempted, no microcode; SMT Host state unknown
Vulnerability Retbleed:             Not affected
Vulnerability Spec store bypass:    Mitigation; Speculative Store Bypass disabled via prctl and seccomp
Vulnerability Spectre v1:           Mitigation; usercopy/swapgs barriers and __user pointer sanitization
Vulnerability Spectre v2:           Mitigation; Enhanced IBRS, IBPB conditional, RSB filling, PBRSB-eIBRS SW sequence
Vulnerability Srbds:                Not affected
Vulnerability Tsx async abort:      Not affected
Flags:                              fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush mmx fxsr sse sse2 ss ht syscall nx pdpe1gb rdtscp lm constant_tsc rep_good nopl xtopology nonstop_tsc cpuid tsc_known_freq pni pclmulqdq monitor ssse3 fma cx16 pdcm pcid sse4_1 sse4_2 x2apic movbe popcnt aes xsave avx f16c rdrand hypervisor lahf_lm abm 3dnowprefetch cpuid_fault invpcid_single ssbd ibrs ibpb stibp ibrs_enhanced fsgsbase tsc_adjust bmi1 avx2 smep bmi2 erms invpcid avx512f avx512dq rdseed adx smap avx512ifma clflushopt clwb avx512cd sha_ni avx512bw avx512vl xsaveopt xsavec xgetbv1 xsaves wbnoinvd arat avx512vbmi avx512_vbmi2 gfni vaes vpclmulqdq avx512_vnni avx512_bitalg avx512_vpopcntdq rdpid arch_capabilities

Versions of relevant libraries:
[pip3] numpy==1.26.1
[pip3] torch==2.1.0
[pip3] torchaudio==2.1.0
[pip3] torchvision==0.16.0
[pip3] triton==2.1.0
[conda] Could not collect

--------------------------------nvidia-smi---------------------------------------------
![1697625105315](https://github.com/pytorch/pytorch/assets/45593232/f0ff9e72-f419-46c3-8df3-fd1d83e1f297)

--------------------------------cuda version---------------------------------------------
![1697625161049](https://github.com/pytorch/pytorch/assets/45593232/46c66f85-c3c5-4439-bf3d-f019d3eead03)

--------------------------------install torch command---------------------------------------------
pip3 install torch torchvision torchaudio

--------------------------------python lib---------------------------------------------
certifi                  2019.11.28
chardet                  3.0.4
command-not-found        0.3
dbus-python              1.2.16
distro                   1.4.0
distro-info              0.23+ubuntu1.1
filelock                 3.12.4
fsspec                   2023.9.2
idna                     2.8
Jinja2                   3.1.2
language-selector        0.1
MarkupSafe               2.1.3
mpmath                   1.3.0
netifaces                0.10.4
networkx                 3.1
numpy                    1.26.1
nvidia-cublas-cu12       12.1.3.1
nvidia-cuda-cupti-cu12   12.1.105
nvidia-cuda-nvrtc-cu12   12.1.105
nvidia-cuda-runtime-cu12 12.1.105
nvidia-cudnn-cu12        8.9.2.26
nvidia-cufft-cu12        11.0.2.54
nvidia-curand-cu12       10.3.2.106
nvidia-cusolver-cu12     11.4.5.107
nvidia-cusparse-cu12     12.1.0.106
nvidia-nccl-cu12         2.18.1
nvidia-nvjitlink-cu12    12.2.140
nvidia-nvtx-cu12         12.1.105
Pillow                   10.1.0
pip                      23.3
PyGObject                3.36.0
pymacaroons              0.13.0
PyNaCl                   1.3.0
python-apt               2.0.1+ubuntu0.20.4.1
PyYAML                   5.3.1
requests                 2.22.0
requests-unixsocket      0.2.0
setuptools               45.2.0
six                      1.14.0
ssh-import-id            5.10
sympy                    1.12
torch                    2.1.0
torchaudio               2.1.0
torchvision              0.16.0
triton                   2.1.0
typing_extensions        4.8.0
ubuntu-advantage-tools   8001
ufw                      0.36
unattended-upgrades      0.1
urllib3                  1.25.8
wheel                    0.34.2

**Comment by ptrblck** [comment]
Not reproducible using `pip install torch torchvision torchaudio`:
```python
pip install torch torchvision torchaudio
Looking in indexes: https://pypi.org/simple, https://pypi.ngc.nvidia.com
Collecting torch
  Downloading torch-2.1.0-cp310-cp310-manylinux1_x86_64.whl.metadata (25 kB)
Requirement already satisfied: torchvision in /usr/local/lib/python3.10/dist-packages (0.16.0a0)
Collecting torchaudio
  Downloading torchaudio-2.1.0-cp310-cp310-manylinux1_x86_64.whl.metadata (5.7 kB)
Requirement already satisfied: filelock in /usr/local/lib/python3.10/dist-packages (from torch) (3.12.4)
Requirement already satisfied: typing-extensions in /usr/local/lib/python3.10/dist-packages (from torch) (4.7.1)
Requirement already satisfied: sympy in /usr/local/lib/python3.10/dist-packages (from torch) (1.12)
Requirement already satisfied: networkx in /usr/local/lib/python3.10/dist-packages (from torch) (2.6.3)
Requirement already satisfied: jinja2 in /usr/local/lib/python3.10/dist-packages (from torch) (3.1.2)
Requirement already satisfied: fsspec in /usr/local/lib/python3.10/dist-packages (from torch) (2023.6.0)
Collecting nvidia-cuda-nvrtc-cu12==12.1.105 (from torch)
  Downloading nvidia_cuda_nvrtc_cu12-12.1.105-py3-none-manylinux1_x86_64.whl (23.7 MB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 23.7/23.7 MB 27.2 MB/s eta 0:00:00
Collecting nvidia-cuda-runtime-cu12==12.1.105 (from torch)
  Downloading nvidia_cuda_runtime_cu12-12.1.105-py3-none-manylinux1_x86_64.whl (823 kB)
...
```
```python
# python -c "import torch; print(torch.__version__); print(torch.__path__)"
2.1.0+cu121
['/usr/local/lib/python3.10/dist-packages/torch']
# find /usr/ -name libnvJit*
/usr/local/lib/python3.10/dist-packages/nvidia/nvjitlink/lib/libnvJitLink.so.12
```

**Comment by qianxifu** [comment]
thanks for your help.

**Comment by lee101** [comment]
Note that ptrblock is on cuda 12.1 and we are having this issue on cuda 12.0
Not sure if thats related but worth a try updating cuda versions

**Comment by lee101** [comment]
LD_LIBRARY_PATH=/usr/local/cuda-11.7/lib64

Managed to get this error message to go away by pointing this env var back to a previous cuda version (11.7 in my case instead of 12.0), not sure what this was about.

-Lee text-generator.io

**Comment by panpan0000** [comment]
> I had the same problem of this issue: `File "/home/user/.conda/envs/myenv/bin/torchrun", line 5, in <module> from torch.distributed.run import main File "/home/user/.conda/envs/myenv/lib/python3.8/site-packages/torch/__init__.py", line 235, in <module> from torch._C import * # noqa: F403 ImportError: /home/user/.conda/envs/myenv/lib/python3.8/site-packages/torch/lib/../../nvidia/cusparse/lib/libcusparse.so.12: undefined symbol: __nvJitLinkAddData_12_1, version libnvJitLink.so.12`
> 
> I tried the solution suggested by @ptrblck and the error is not fixed, yet!

which solution ? I saw the same issue with below trial (python3.10)
```
pip3 install torch torchvision torchaudio

pip list |grep torch
torch                     2.1.0
torchaudio                2.1.0
torchvision               0.16.0
```
```
python3 -c "import torch;"

from torch._C import *  # noqa: F403
ImportError: /usr/local/lib/python3.10/dist-packages/torch/lib/../../nvidia/cusparse/lib/libcusparse.so.12: undefined symbol: __nvJitLinkAddData_12_1, version libnvJitLink.so.12
```

**Comment by kk19990709** [comment]
I don't think this issue should be closed. It hasn't been solved yet. Same Error within torch==2.1.0

**Comment by kk19990709** [comment]
> > I had the same problem of this issue: `File "/home/user/.conda/envs/myenv/bin/torchrun", line 5, in <module> from torch.distributed.run import main File "/home/user/.conda/envs/myenv/lib/python3.8/site-packages/torch/__init__.py", line 235, in <module> from torch._C import * # noqa: F403 ImportError: /home/user/.conda/envs/myenv/lib/python3.8/site-packages/torch/lib/../../nvidia/cusparse/lib/libcusparse.so.12: undefined symbol: __nvJitLinkAddData_12_1, version libnvJitLink.so.12`我遇到了同样的问题： `File "/home/user/.conda/envs/myenv/bin/torchrun", line 5, in <module> from torch.distributed.run import main File "/home/user/.conda/envs/myenv/lib/python3.8/site-packages/torch/__init__.py", line 235, in <module> from torch._C import * # noqa: F403 ImportError: /home/user/.conda/envs/myenv/lib/python3.8/site-packages/torch/lib/../../nvidia/cusparse/lib/libcusparse.so.12: undefined symbol: __nvJitLinkAddData_12_1, version libnvJitLink.so.12` 
> > I tried the solution suggested by @ptrblck and the error is not fixed, yet!我尝试了建议的解决方案，但错误尚未修复！
> 
> May I ask which solution ? I saw the same issue with below trial ( python version == 3.10)请问哪种解决方案？我在下面的试用版（python版本== 3.10）中看到同样的问题
> 
> Below is my env setup and issue:以下是我的环境设置和问题：
> 
> ```
> pip3 install torch torchvision torchaudio
> 
> pip list |grep torch
> torch                     2.1.0
> torchaudio                2.1.0
> torchvision               0.16.0
> 
> cat /proc/driver/nvidia/version
> NVRM version: NVIDIA UNIX x86_64 Kernel Module  525.105.17  Tue Mar 28 18:02:59 UTC 2023
> GCC version:  gcc version 11.4.0 (Ubuntu 11.4.0-1ubuntu1~22.04)
> ```
> 
> ```
> python3 -c "import torch;"
> 
> from torch._C import *  # noqa: F403
> ImportError: /usr/local/lib/python3.10/dist-packages/torch/lib/../../nvidia/cusparse/lib/libcusparse.so.12: undefined symbol: __nvJitLinkAddData_12_1, version libnvJitLink.so.12
> ```
> 
> Workaround: 解决方法：
> 
> When downgrade torch to 2.0.1 (`pip3 install torch==2.0.1`), issue gone.将割炬降级到 2.0.1 （ `pip3 install torch==2.0.1` ） 时，问题消失了。

This works, but some package require torch==2.1.0, such as xformers

**Comment by upenn-hughmac** [comment]
Same issue. In case it's useful for others, fixed for me by either:
```
export LD_LIBRARY_PATH=$HOME/path/to/my/venv3115/lib64/python3.11/site-packages/nvidia/nvjitlink/lib:$LD_LIBRARY_PATH
```
or uninstalling 2.1.0 (stable) and [installing the nightly dev preview](https://pytorch.org/get-started/locally/):
```
python -m pip uninstall torch torchvision torchaudio
python -m pip install --pre torch torchvision torchaudio --index-url https://download.pytorch.org/whl/nightly/cu121
```

**Comment by conan1024hao** [comment]
> When downgrade torch to 2.0.1 (pip3 install torch==2.0.1), issue gone.
torch wasn't the problem to me, downgrade torch audio to 2.0.1, issue gone.

**Comment by krodio** [comment]
> Same issue. In case it's useful for others, fixed for me by either:
> 
> ```
> export LD_LIBRARY_PATH=$HOME/path/to/my/venv3115/lib64/python3.11/site-packages/nvidia/nvjitlink/lib:$LD_LIBRARY_PATH
> ```
> 
> or uninstalling 2.1.0 (stable) and [installing the nightly dev preview](https://pytorch.org/get-started/locally/):
> 
> ```
> python -m pip uninstall torch torchvision torchaudio
> python -m pip install --pre torch torchvision torchaudio --index-url https://download.pytorch.org/whl/nightly/cu121
> ```

it did work, replace xxxx to the real python interpreter path
export LD_LIBRARY_PATH=$HOME/xxxx/python3.11/site-packages/nvidia/nvjitlink/lib:$LD_LIBRARY_PATH

**Comment by yoyo20010808** [comment]
pip uninstall torch torchvision torchaudio

**Comment by NintendoLink** [comment]
it's work,thanks~ @upenn-hughmac

**Comment by surmount1** [comment]
> Same issue. In case it's useful for others, fixed for me by either:
> 
> ```
> export LD_LIBRARY_PATH=$HOME/path/to/my/venv3115/lib64/python3.11/site-packages/nvidia/nvjitlink/lib:$LD_LIBRARY_PATH
> ```
> 
> or uninstalling 2.1.0 (stable) and [installing the nightly dev preview](https://pytorch.org/get-started/locally/):
> 
> ```
> python -m pip uninstall torch torchvision torchaudio
> python -m pip install --pre torch torchvision torchaudio --index-url https://download.pytorch.org/whl/nightly/cu121
> ```

Nice!  I also solved this problem using this method.
export LD_LIBRARY_PATH=/data/home/user/anaconda3/envs/vllm/lib/python3.10
/site-packages/nvidia/nvjitlink/lib:$LD_LIBRARY_PATH

**Comment by jianbo27** [comment]
I try to downgrade to python3.9, which works for me in conda virtual environment.

**Comment by wangbluo** [comment]
Same issue on torch2.2, I have tried all above solutions but failed, this issue shouldn't be closed at all.

python version: python 3.10
PyTorch-cuda:12.1
$CUDA_HOME: /home/share/spack/opt/spack/linux-ubuntu20.04-icelake/gcc-9.4.0/cuda-12.1.1-uxo2fr2s3d6ge4m6bo46jslallfbluei

**Comment by osma** [comment]
I've also had this problem. In my case, it was apparently due to a compatibility issue w.r.t. CUDA 12.0.0 that I was using.

It appears that PyTorch 2.1.x and 2.2.0 have been compiled against CUDA 12.1.0 and they use new symbols introduced in 12.1 so they won't work with CUDA 12.0.0. Installing either CUDA 12.1.0 or the older version 11.8.0 fixes the problem for me. Downgrading to PyTorch 2.0.1 also works, as it's compatible with CUDA 12.0.0.

**Comment by yangfansun** [comment]
Installing PyTorch with the official CUDA 11.8 setup recommended by PyTorch can fix this problem.

**Comment by richardp4** [comment]
Hi,
I got the same problem.
My conditions are below.
- OS : ubuntu 22.04
- CUDA : 12.0
- cudnn : 8.8
- python : 3.9 anaconda env
- pytorch : 2.2.2 -> 2.0.1 (down grade)

After downgrading the pytorch version from 2.2.2 to 2.0.1, import torch is good.
However, another error occurred like "ModuleNotFoundError: No module named 'torch._custom_ops'"

Please give me some tips to solve it.

**Comment by osma** [comment]
@richardp4 CUDA version 12.0 is your problem. Either upgrade it to 12.1+ or downgrade to 11.8.

**Comment by M3Dade** [comment]
> downgrade to 11.8.

@osma  I meet the problem by  downloading flash_attn-2.5.7.tar.gz. It's worked when I downgrade CUDA version from 12.0 to 11.8.  Thanks for your suggestion.

**Comment by weifengpy** [comment]
> Same issue. In case it's useful for others, fixed for me by either:
> 
> ```
> export LD_LIBRARY_PATH=$HOME/path/to/my/venv3115/lib64/python3.11/site-packages/nvidia/nvjitlink/lib:$LD_LIBRARY_PATH
> ```
> 
> or uninstalling 2.1.0 (stable) and [installing the nightly dev preview](https://pytorch.org/get-started/locally/):
> 
> ```
> python -m pip uninstall torch torchvision torchaudio
> python -m pip install --pre torch torchvision torchaudio --index-url https://download.pytorch.org/whl/nightly/cu121
> ```

the quoted answer works for me. in case people curious about how to find `site-packages` path
step 1: `python3 -m pip list -v`: this shows full path 'envs/pytorch-3.10/lib/python3.10/site-package'
step 2: `export LD_LIBRARY_PATH=full/path/envs/pytorch-3.10/lib/python3.10/site-packages/nvidia/nvjitlink/lib:$LD_LIBRARY_PATH`

**Comment by peterzsj6** [comment]
> Same issue. In case it's useful for others, fixed for me by either:
> 
> ```
> export LD_LIBRARY_PATH=$HOME/path/to/my/venv3115/lib64/python3.11/site-packages/nvidia/nvjitlink/lib:$LD_LIBRARY_PATH
> ```
> 
> or uninstalling 2.1.0 (stable) and [installing the nightly dev preview](https://pytorch.org/get-started/locally/):
> 
> ```
> python -m pip uninstall torch torchvision torchaudio
> python -m pip install --pre torch torchvision torchaudio --index-url https://download.pytorch.org/whl/nightly/cu121
> ```

I also fixed it by updating my python version from 3.8 to 3.10

**Comment by redmiyyds** [comment]
> > 同样的问题。如果对其他人有用，可以通过以下任一方式为我修复：
> > ```
> > export LD_LIBRARY_PATH=$HOME/path/to/my/venv3115/lib64/python3.11/site-packages/nvidia/nvjitlink/lib:$LD_LIBRARY_PATH
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > 或者卸载 2.1.0 (稳定版) 并[安装夜间开发预览版](https://pytorch.org/get-started/locally/)：
> > ```
> > python -m pip uninstall torch torchvision torchaudio
> > python -m pip install --pre torch torchvision torchaudio --index-url https://download.pytorch.org/whl/nightly/cu121
> > ```
> 
> 它确实有效，将 xxxx 替换为真正的 python 解释器路径 export LD_LIBRARY_PATH=$HOME/xxxx/python3.11/site-packages/nvidia/nvjitlink/lib:$LD_LIBRARY_PATH

请问替换的这个路径是哪个啊，我有不同的环境，具体是哪的呀

**Comment by haofanwang** [comment]
I solved following this [blog](https://zenn.dev/ykesamaru/articles/9daff09bed17ba), but remember, after export, you need to run `source ~/.bashrc` to make to valid.

**Comment by JakeCirino** [comment]
> @richardp4 CUDA version 12.0 is your problem. Either upgrade it to 12.1+ or downgrade to 11.8.

Was having this issue trying to run a Mistral model on a Paperspace notebook. Downgrading to 11.8 was the solution for me:

`python -m pip uninstall torch torchvision torchaudio
pip3 install torch torchvision torchaudio --index-url https://download.pytorch.org/whl/cu118`

**Comment by freckletonj** [comment]
This just worked for me:

* reinstall latest cuda: https://developer.nvidia.com/cuda-downloads
* restart
* :

```sh
pip uninstall torch torchvision
pip install --no-cache-dir torch torchvision
```

**Comment by MedicineStick** [comment]
> Same issue. In case it's useful for others, fixed for me by either:
> 
> ```
> export LD_LIBRARY_PATH=$HOME/path/to/my/venv3115/lib64/python3.11/site-packages/nvidia/nvjitlink/lib:$LD_LIBRARY_PATH
> ```
> 
> or uninstalling 2.1.0 (stable) and [installing the nightly dev preview](https://pytorch.org/get-started/locally/):
> 
> ```
> python -m pip uninstall torch torchvision torchaudio
> python -m pip install --pre torch torchvision torchaudio --index-url https://download.pytorch.org/whl/nightly/cu121
> ```

It works on torch 2.5.1, thank you

**Comment by yshuolu** [comment]
> LD_LIBRARY_PATH=/usr/local/cuda-11.7/lib64
> 
> Managed to get this error message to go away by pointing this env var back to a previous cuda version (11.7 in my case instead of 12.0), not sure what this was about, also works if i just unset that variable so i'm not sure if we need to set that up with cuda 12.0 or not.
> 
> -Lee https://text-generator.io

This magically works for me, among all the answers.

**Comment by RahulGottimukkula-kore** [comment]
I am facing the same error. The above mentioned solution works but I have to repeat it everytime. Can anyone tell me why this particular issue is occuring. Is there no solution to it ?

**Comment by tmct** [comment]
Yes, it would be very good to see this work out of the box for 12.0

**Comment by vTuanpham** [comment]
Hoping a fix is coming soon!

**Comment by XuehaoSun** [comment]
> 请问替换的这个路径是哪个啊，我有不同的环境，具体是哪的呀

If you want to get the path automatically:
`export LD_LIBRARY_PATH=$(which python)$(python -c 'import sys; print(".".join(map(str, sys.version_info[:2])))')/site-packages/nvidia/nvjitlink/lib:$LD_LIBRARY_PATH`

**Comment by qgallouedec** [comment]
Solve with

```
export LD_LIBRARY_PATH=$(python -c "import site; print(site.getsitepackages()[0] + '/nvidia/nvjitlink/lib')"):$LD_LIBRARY_PATH
```

I think this issue should be re-opened

**Comment by hafezmg48** [comment]
This issue still persists with the torch 2.5.1 and :  NVIDIA-SMI 545.23.08              Driver Version: 545.23.08    CUDA Version: 12.3 

I had to downgrade to torch 2.0.1 to make it work. They don't plan to fix it?

**Comment by khanfarhan10** [comment]
@qianxifu we are still facing this issue, is there any update on this?

Can you please look into it if possible?

**Comment by FurkanGozukara** [comment]
On kaggle if you install any torch version it breaks everything. therefore we are 100% stuck whatever version they use

**Comment by hecatolite** [comment]
> Same issue. In case it's useful for others, fixed for me by either:
> 
> ```
> export LD_LIBRARY_PATH=$HOME/path/to/my/venv3115/lib64/python3.11/site-packages/nvidia/nvjitlink/lib:$LD_LIBRARY_PATH
> ```
> 
> or uninstalling 2.1.0 (stable) and [installing the nightly dev preview](https://pytorch.org/get-started/locally/):
> 
> ```
> python -m pip uninstall torch torchvision torchaudio
> python -m pip install --pre torch torchvision torchaudio --index-url https://download.pytorch.org/whl/nightly/cu121
> ```

I have tried to set the PATH and a nightly dev, but they didn't work.
Similarly, I try to change my torch to 11.8, then solve the question.
My cuda version is 12.0.

**Comment by enerai** [comment]
> > > 同样的问题。如果对其他人有用，可以通过以下任一方式为我修复：
> > > ```
> > > export LD_LIBRARY_PATH=$HOME/path/to/my/venv3115/lib64/python3.11/site-packages/nvidia/nvjitlink/lib:$LD_LIBRARY_PATH
> > > ```
> > > 
> > > 
> > >     
> > >       
> > >     
> > > 
> > >       
> > >     
> > > 
> > >     
> > >   
> > > 或者卸载 2.1.0 (稳定版) 并[安装夜间开发预览版](https://pytorch.org/get-started/locally/)：
> > > ```
> > > python -m pip uninstall torch torchvision torchaudio
> > > python -m pip install --pre torch torchvision torchaudio --index-url https://download.pytorch.org/whl/nightly/cu121
> > > ```
> > 
> > 
> > 它确实有效，将 xxxx 替换为真正的 python 解释器路径 export LD_LIBRARY_PATH=$HOME/xxxx/python3.11/site-packages/nvidia/nvjitlink/lib:$LD_LIBRARY_PATH
> 
> 请问替换的这个路径是哪个啊，我有不同的环境，具体是哪的呀

用这个获取 $(conda info --base)

**Comment by iseesaw** [comment]
The following command solved my problem (vllm 0.6.4 + torch 2.5.1 + cuda 12.1)

```
pip install patchelf
patchelf --add-rpath '$ORIGIN/../../nvidia/nvjitlink/lib' /root/miniconda3/envs/openrlhf/lib/python3.10/site-packages/torch/lib/libtorch_cuda.so
```

Ref to https://github.com/pytorch/pytorch/issues/140797#issuecomment-2486771624

**Comment by LukeLIN-web** [comment]
Meet similar problem `envs/camel/lib/python3.10/site-packages/torch/lib/../../nvidia/cusparse/lib/libcusparse.so.12: undefined symbol: __nvJitLinkComplete_12_4, version libnvJitLink.so.12`

**Comment by LukeLIN-web** [comment]
In my case, this is because my cuda 12.3<  12.4 in latest torch.

So reinstall the torch`pip install torch==2.4.1 torchvision==0.19.1 torchaudio==2.4.1 --index-url https://download.pytorch.org/whl/cu121` can solve my probhlem. 

So maybe https://github.com/pytorch/pytorch/issues/111469#issue-1949414420  `CUDA runtime version: 12.0.140` also because 12.0  < __nvJitLinkAddData_12_1, ver.

**Comment by jxmorris12** [comment]
This is incredibly frustrating. I uninstalled everything and then installed with conda, with the proper flags set -- for me:
```bash
conda install pytorch torchvision torchaudio pytorch-cuda=12.4 -c pytorch -c nvidia --force-reinstall
```
and it worked. Pip/uv didn't work though.

**Comment by fanfouff** [comment]
`conda install python==3.8`maybe work,I used python3.9 has the same error

**Comment by joe32140** [comment]
Got a similar issue with python 3.11. It works for me if I downgrade to `pip install torch==2.4.0`.

**Comment by OFSkean** [comment]
Just had this with Python 3.10.6 and Torch 2.5.1. Like the others, downgrading the torch version helped

**Comment by xiejibing** [comment]
> This issue still persists with the torch 2.5.1 and : NVIDIA-SMI 545.23.08 Driver Version: 545.23.08 CUDA Version: 12.3
> 
> I had to downgrade to torch 2.0.1 to make it work. They don't plan to fix it?

Just encountered the same issue. With cuda 12.2, torch 2.5.1.

**Comment by hafezmg48** [comment]
> Just had this with Python 3.10.6 and Torch 2.5.1. Like the others, downgrading the torch version helped

Is it possible to reopen this issue?

**Comment by nebulaanish** [comment]
The easy fix is to not pin torch to a specific version, but allow torchvision or torchaudio to install the appropriate version that they're dependent on. I came across the same issue when trying to pin the version of both `pytorch`, `cudatoolkit` and `pytorch-cuda`. 

Allowing `pytorch-cuda=11.8` to install the `pytorch` by itself, solved this issue for me.

**Comment by wizyoung** [comment]
Config cuda12.4 for torch==2.5.1 or just set `unset LD_LIBRARY_PATH`

**Comment by murfel** [comment]
Same bug on macOS Sonoma 14.6.1. Downgrading pytorch from latest (2.5.1) to `2.2.0` helped.

**Comment by qq280948982** [comment]
> Same issue. In case it's useful for others, fixed for me by either:
> 
> ```
> export LD_LIBRARY_PATH=$HOME/path/to/my/venv3115/lib64/python3.11/site-packages/nvidia/nvjitlink/lib:$LD_LIBRARY_PATH
> ```
> 
> or uninstalling 2.1.0 (stable) and [installing the nightly dev preview](https://pytorch.org/get-started/locally/):
> 
> ```
> python -m pip uninstall torch torchvision torchaudio
> python -m pip install --pre torch torchvision torchaudio --index-url https://download.pytorch.org/whl/nightly/cu121
> ```

you save my day !!!

**Comment by wfloveiu** [comment]
> Config cuda12.4 for torch==2.5.1 or just set `unset LD_LIBRARY_PATH`

`unset LD_LIBRARY_PATH` works for me!

**Comment by HHTao16** [comment]
> Config cuda12.4 for torch==2.5.1 or just set `unset LD_LIBRARY_PATH`

`unset LD_LIBRARY_PATH` works for me!

**Comment by Jianguo99** [comment]
> Config cuda12.4 for torch==2.5.1 or just set `unset LD_LIBRARY_PATH`为 torch==2.5.1 配置 cuda12.4 或仅设置 `unset LD_LIBRARY_PATH`

Cool! It works for me. My config is torch==2.5.1 and cuda12.6

**Comment by Jaykumaran** [comment]
Was having same issue with global cuda 12.2.

Env: torch = 2.5.1 cu 12.4

Fixed by downgrading to cu 12.1

conda install pytorch==2.5.1 torchvision==0.20.1 torchaudio==2.5.1 pytorch-cuda=12.1 -c pytorch -c nvidia

**Comment by hychaochao** [comment]
> > Same issue. In case it's useful for others, fixed for me by either:
> > ```
> > export LD_LIBRARY_PATH=$HOME/path/to/my/venv3115/lib64/python3.11/site-packages/nvidia/nvjitlink/lib:$LD_LIBRARY_PATH
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > or uninstalling 2.1.0 (stable) and [installing the nightly dev preview](https://pytorch.org/get-started/locally/):
> > ```
> > python -m pip uninstall torch torchvision torchaudio
> > python -m pip install --pre torch torchvision torchaudio --index-url https://download.pytorch.org/whl/nightly/cu121
> > ```
> 
> the quoted answer works for me. in case people curious about how to find `site-packages` path step 1: `python3 -m pip list -v`: this shows full path 'envs/pytorch-3.10/lib/python3.10/site-package' step 2: `export LD_LIBRARY_PATH=full/path/envs/pytorch-3.10/lib/python3.10/site-packages/nvidia/nvjitlink/lib:$LD_LIBRARY_PATH`

nice! resolved!
